### PR TITLE
vkjson: Fix include paths and clang errors

### PR DESCRIPTION
**Pull request by critsec**

### CODE REVIEW

**Comment by courtney-g** [comment]
+1

File: libs/vkjson/vkjson.cc [hunk 1]
```diff
@@ -456,7 +456,7 @@ inline void VisitForWrite(Visitor* visitor, const T& t) {
   Iterate(visitor, const_cast<T*>(&t));
 }
 
-template <typename T, typename = EnableForStruct<T>, typename = void>
+template <typename T, typename /*= EnableForStruct<T>*/, typename /*= void*/>
```
Comment (pimanttr) [inline libs/vkjson/vkjson.cc]: EnableForStruct is needed for SFINAE - to make sure this overload is only used if T is a struct. I'm not sure how this compiles without it - nor what problem it causes on clang?

File: libs/vkjson/vkjson.cc [hunk 2]
```diff
@@ -634,7 +634,7 @@ class JsonReaderVisitor {
   std::string* errors_;
 };
 
-template <typename T, typename = EnableForStruct<T>>
+template <typename T, typename /*= EnableForStruct<T>*/>
```
Comment (pimanttr) [inline libs/vkjson/vkjson.cc]: (same here)

**Comment by critsec** [comment]
The problem is that the default value is stated twice, clang is issuing an error about redefining template default values. For line 459, see line 361.

**Comment by pimanttr** [comment]
Ah, ok, got it, the forward declarations already have it so it's fine to remove in the definition.

+1 then

**Pull request merged**